service/server/router_linux.cpp

```diff
@@ -170,3 +175,122 @@ void RouterLinux::flushDns()
     else
         qDebug().noquote() << "OUTPUT systemctl restart nscd/systemd-resolved: " + output;
 }
+
+bool RouterLinux::createTun(const QString &dev, const QString &subnet) {
+    qDebug().noquote() << "createTun start";
+
+    char cmd [1000] = {0x0};
+    sprintf(cmd, "ip tuntap add mode tun dev %s", dev.toStdString().c_str());
+    int sys = system(cmd);
```

Comment:
system() creates new shell process for executing commands.
It's not safe to use, bc it may stuck service.
It's better to rewrite using QProcess, like
  ```
 QProcess process;
    QStringList commands;

    commands << "ip" << "tuntap" << "add" << "mode" << "tun" << "dev" << dev;
    process.start("sudo", commands);
```
and wait some time for process finish